internal/service/fsx/openzfs_file_system.go

```diff
@@ -0,0 +1,844 @@
+	d.Set("daily_automatic_backup_start_time", openzfsConfig.DailyAutomaticBackupStartTime)
+	d.Set("copy_tags_to_backups", openzfsConfig.CopyTagsToBackups)
+	d.Set("copy_tags_to_volumes", openzfsConfig.CopyTagsToVolumes)
+
+	rootVolume, err := FindVolumeByID(conn, *openzfsConfig.RootVolumeId)
+
+	if err != nil {
+		return fmt.Errorf("error reading FSx OpenZFS Root Volume Configuration (%s): %w", *openzfsConfig.RootVolumeId, err)
+	}
+
+	if err := d.Set("root_volume_configuration", flattenFsxOpenzfsRootVolumeConfiguration(rootVolume)); err != nil {
+		return fmt.Errorf("error setting root_volume_configuration: %w", err)
+	}
+
+	return nil
+}
+
+func resourceOpenzfsFileSystemUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).FSxConn
+
+	if d.HasChange("tags_all") {
+		o, n := d.GetChange("tags_all")
+
+		if err := UpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return fmt.Errorf("error updating FSx OpenZFS File System (%s) tags: %w", d.Get("arn").(string), err)
+		}
+	}
+
+	if d.HasChangesExcept("tags_all", "tags") {
+		input := &fsx.UpdateFileSystemInput{
+			ClientRequestToken:   aws.String(resource.UniqueId()),
+			FileSystemId:         aws.String(d.Id()),
+			OpenZFSConfiguration: &fsx.UpdateFileSystemOpenZFSConfiguration{},
+		}
+
+		if d.HasChange("automatic_backup_retention_days") {
+			input.OpenZFSConfiguration.AutomaticBackupRetentionDays = aws.Int64(int64(d.Get("automatic_backup_retention_days").(int)))
+		}
+
+		if d.HasChange("copy_tags_to_backups") {
+			input.OpenZFSConfiguration.CopyTagsToBackups = aws.Bool(d.Get("copy_tags_to_backups").(bool))
+		}
+
+		if d.HasChange("copy_tags_to_volumes") {
+			input.OpenZFSConfiguration.CopyTagsToVolumes = aws.Bool(d.Get("copy_tags_to_volumes").(bool))
+		}
+
+		if d.HasChange("daily_automatic_backup_start_time") {
+			input.OpenZFSConfiguration.DailyAutomaticBackupStartTime = aws.String(d.Get("daily_automatic_backup_start_time").(string))
+		}
+
+		if d.HasChange("disk_iops_configuration") {
+			input.OpenZFSConfiguration.DiskIopsConfiguration = expandFsxOpenzfsFileDiskIopsConfiguration(d.Get("throughput_capacity").([]interface{}))
+		}
+
+		if d.HasChange("throughput_capacity") {
+			input.OpenZFSConfiguration.ThroughputCapacity = aws.Int64(int64(d.Get("throughput_capacity").(int)))
+		}
+
+		if d.HasChange("weekly_maintenance_start_time") {
+			input.OpenZFSConfiguration.WeeklyMaintenanceStartTime = aws.String(d.Get("weekly_maintenance_start_time").(string))
+		}
+
+		_, err := conn.UpdateFileSystem(input)
+
+		if err != nil {
+			return fmt.Errorf("error updating FSx OpenZFS File System (%s): %w", d.Id(), err)
+		}
+
+		if _, err := waitFileSystemUpdated(conn, d.Id(), d.Timeout(schema.TimeoutUpdate)); err != nil {
+			return fmt.Errorf("error waiting for FSx OpenZFS File System (%s) update: %w", d.Id(), err)
+		}
+
+		if d.HasChange("root_volume_configuration") {
```

Comment:
worth extract to be on the same level as fs and tags as seems you can update volume without updating the fs itself

Reply:
I will make a separate volume resource for openzfs just like ontap but this is for the root volume. via the API you can set the root volume properties in the createfilesystem API but can't read or update it with the file system APIs so need this split logic.